### ISSUE TITLE
0.1.0: DNS seems not working from pods

### ISSUE THREAD

**Issue by mumoshu** [kubernetes/minikube#137]
Hi, thanks for developing minikube :)

I have just tested 0.1.0 and realized that DNS isn't working inside pods/containers.

Here's how I have tested it:

```
$ helm install alpine
$ kubectl exec -it alpine /bin/sh -- -c 'nslookup kubernetes'
Server:    (null)
Address 1: ::1 localhost
Address 2: 127.0.0.1 localhost

nslookup: can't resolve 'kubernetes': Try again
error: error executing remote command: error executing command in container: Error executing in Docker Container: 1
```

```
$ kubectl exec -it alpine /bin/sh -- -c 'nslookup github.com'
Server:    (null)
Address 1: ::1 localhost
Address 2: 127.0.0.1 localhost

error: error executing remote command: error executing command in container: Error executing in Docker Container: 1
```

I don't know if this is known issue for 0.1.0 but anyway I'm leaving this.

**Comment by dlorenc** [comment]
Thanks, something definitely seems to not be working here. I ran the same and can verify there's an error. Here are some relevant logs:
```
 ./out/minikube logs | grep kube2sky
I0531 15:29:02.851275    4791 kube2sky.go:486] Etcd server found: http://localhost:9090
I0531 15:29:03.852716    4791 kube2sky.go:553] Using http://127.0.0.1:8080 for kubernetes master
I0531 15:29:03.852734    4791 kube2sky.go:554] Using kubernetes API v1
I0531 15:29:03.852875    4791 kube2sky.go:622] Waiting for service: default/kubernetes
I0531 15:29:03.855235    4791 kube2sky.go:667] Successfully added DNS record for Kubernetes service.
```

So it appears DNS thinks it's running, but the pod can't connect for some reason.

**Comment by dlorenc** [comment]
I don't see the service or endpoint:
```
$ kubectl get services --all-namespaces
NAMESPACE     NAME                   CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
default       kubernetes             10.0.0.1     <none>        443/TCP   1d
kube-system   kubernetes-dashboard   10.0.0.198   <none>        80/TCP    1d
$ kubectl get endpoints --all-namespaces
NAMESPACE     NAME                   ENDPOINTS         AGE
default       kubernetes             10.0.2.15:443     1d
kube-system   kubernetes-dashboard   172.17.0.2:9090   1d
```

cc @ethernetdan Have you ever seen this before? I stuck some logging in dns.go and it looks like the call to Services.Create is succeeding...

**Comment by luxas** [comment]
This is why I filed #114.

If no endpoints or services are present, it won't work; even if the DNS goroutines are running in localkube.
So debugging Services.Create and the like is the way to go...

**Comment by dlorenc** [comment]
Current theory: the addon-manager is deleting the kube-dns service and endpoint because they're getting created in the kube-system namespace.

**Comment by dlorenc** [comment]
This should be fixed with #143. It'll go out in the 0.2.0 release which I'm cutting Friday.
